src/main/java/Duke.java

```diff
@@ -1,10 +1,166 @@
+        while (!userInput.toLowerCase().equals("bye")){
+            String echos = String.format("%s\t %s\n%s",lineDivider,userInput.toLowerCase(),lineDivider);
+            System.out.println(echos);
+            userInput = sc.nextLine();
+        }
+    }
+
+
+    public static void taskApp(){
+        ArrayList<Task> taskList = new ArrayList<>();
+        while (true){
+            System.out.print("Enter command: ");
+            String userInput = sc.nextLine();
+
+            String commandWord = userInput.split(" ")[0].toLowerCase();
+            String content = userInput.substring(userInput.indexOf(" ") + 1);
+
+            switch (commandWord) {
+                case "todo":
+                    System.out.println("todo");
+                    taskList.add(new Todo(content));
+                    addTaskSuccessMessage(taskList, "\tGot it. I've added this task: ");
+                    break;
+                    
+                case "deadline":
+                    System.out.println("deadline");
+                    String deadlineTask = content.split("/")[0].toLowerCase();
+                    String datelineDateBy = content.split("/")[1].split(" ")[1];
+                    taskList.add(new Deadline(deadlineTask,datelineDateBy ));
+                    addTaskSuccessMessage(taskList, "\tGot it. I've added this deadline: ");
+                    break;
+                    
+                case "event":
+                    System.out.println("event");
+                    String eventTask = content.split("/")[0].toLowerCase();
+                    String eventDateBy = content.split("/")[1];
+                    eventDateBy = eventDateBy.substring(eventDateBy.indexOf(" "));
+                    taskList.add(new Event(eventTask,eventDateBy));
+                    addTaskSuccessMessage(taskList, "\tGot it. I've added this Event: ");
+                    break;
+
+                case "list":
+
+                    String listReturnString = String.format("%s%s%s",lineDivider,getList(taskList),lineDivider);
+                    System.out.println(listReturnString);
+
+                    break;
+
+                case "done":
+                    markTaskDone(taskList, userInput, content);
+
+                    break;
+
+                case "bye":
+                    System.out.println(dukeFarewell);
+                    System.exit(0);
+                default:
+                    System.out.println("Command word not recognised - please start command with " +
+                            "'todo', 'deadline' or 'event'");
+            }
+        }
+
+    }
+
+    public static void markTaskDone(ArrayList<Task> taskList, String userInput, String content) {
+        if (userInput.matches(".*\\d.*")) { // checks if there is a number in done cmd
+            int taskNumber = Integer.parseInt(content) ;
+            int indexOfTaskToBeMarked = taskNumber - 1;
+            if (indexOfTaskToBeMarked < taskList.size()){
```

Comment:
maybe it is just me but maybe you can try to avoid the nested if else loop if possible? (the arrowhead thing)